### DISCUSSION TITLE
What happens when drones go off z level

### DISCUSSION THREAD

**Discussion by FlameArrow57** [stonepillars/goon-flock#201] [Ideas]
Drones will end up going off z level somehow, whether it be being launched somehow (would need to verify), going through a portal, or something similar, and need a way of not counting towards the flock.

My idea would be that they go dormant, almost already coded in. Basically, they won't count toward the flock, go dormant (don't move, don't do anything), with the idea that their connection with the Flock is out of range and so they revert to an offline mode.

Similar would be done for bits.

Other ways could make it so that they don't count towards the flock, but this also gives a more interesting way for flock drones to not count, and make it more themed in them not converting anything to flock on another z level.

**Comment by stonepillars** [comment]
Sa.le steals them. (this is probably a joke)

**Comment by amylizzle** [comment]
Yeah, I agree they should go dormant. Signal out of range.

We should probably also think about what happens when they go dormant and are brought back to the station's Z level, because somebody will do it just to see what happens.

**Reply by FlameArrow57** [reply]
Yeah, I think losing connection to the Flock would mean they wouldn't be automatically reconnected to the Flock if brought back, Flock would need to manually put them back into the Flock, would also be a design thing for multiple Flocks existing.

**Reply by amylizzle** [reply]
What currently happens if you try to interact with/possess a dormant flock? is that handled?

**Reply by FlameArrow57** [reply]
Don't think it is, but think it would be better to make a separate PR for handling dormancy/flock checks and stuff because I don't think the code really handles in many places, and a way of handling how drones are made undormant needs to be made.
